aoai_vs_oai_diffs.md

```diff
@@ -0,0 +1,81 @@
+## Diff between OAI and AOAI
+
+The differences detailed in this document were derived from comparing `{project_root}/.typespec/openapi.yaml` (OAI swagger file) and `Azure/azure-rest-api-spec/specifications/cognitiveservices/data-plane/AzureOpenAI/preview/2024-05-01-preview/inference.yaml` (AOAI latest release version for the SDKs)
+
+## Operations
+
+Listing differences in the request/response models between Azure and non-Azure
+
+### `/completions`
+
+- [x] `CreateCompletionRequest`
+  - [x] Additional Azure field: `completion_config`
+  - [x] non-Azure has `model`
+
+- [ ] `CreateCompletionResponse`
+  - [ ] non-Azure has `system_fingerprint`
+  - [ ] Azure has `prompt_filter_results`
+  - [ ] Azure has under `choices`->`content_filter_results`
+
+
+### `/embeddings`
+
+- [ ] `CreateEmbeddingRequest`
+  - [ ] Azure has `input_type`
+  - [ ] non-Azure has `model`
+
+- [x] `CreateEmbeddingResponse`
+  - [x] Identical
+
+### `/chat/completions`
+- [ ] `CreateChatCompletionRequest`
+  - [ ] Azure has an additional header called `apim-request-id`
+  - [ ] Azure has `data_sources`
+  - [ ] non-Azure has the additional:
+    - [ ] `model`
+    - [ ] `stream_options`
```

Comment:
These two (`stream_options` and `parallel_tool_calls`) are added with `2024-07-01-preview`; that might simplify the initial diff but also leaves the good question of how we handle *version-aware* deltas like this.

Reply:
Yeah, that's somehow something I felt like I needed to have answer for before moving forward. In the absence of a `ServiceApiVersion` enum I also simply assumed that what is supported is `v2024-05-01-preview` from the Azure side.